src/wire/mod.rs

```diff
@@ -91,6 +91,8 @@ mod ipv6option;
 mod icmpv4;
 #[cfg(feature = "proto-ipv6")]
 mod icmpv6;
+#[cfg(feature = "proto-ipv6")]
+mod ndiscoption;
```

Comment:
Why not just `ndisc`? It doesn't matter that it's an option, the protocol is called Neighbor Discovery. Also, you're already dropping `Option` in type names like `NdiscPrefixInfoFlags`.

Reply:
True. This is just the Option parsing, but the actual NDISC parsing will be in `src/wire/icmpv6.rs` per the conversation for #159, so I'll rename it.

Reply:
No, I misunderstood, if this file is just NDISC *option* parsing then this makes sense.

Reply:
:+1:

Reply:
We can actually pull out NDISC into its own `ndisc.rs` file if we mark `Icmpv6Packet::buffer` as `pub(crate)`.

Reply:
So then `NdiscPacket::new` would be something like `NdiscPacket::new(packet: Icmpv6Packet)`? It gets a bit tricky since the type id is in the ICMP header.

Reply:
No, don't add an `NdiscPacket`, just move the `impl` blocks into a separate file.